pandas/io/parsers.py

```diff
@@ -847,12 +847,18 @@ def _validate_usecols_arg(usecols):
     or strings (column by name). Raises a ValueError
     if that is not the case.
     """
+    msg = ("The elements of 'usecols' must "
+           "either be all strings or all integers")
+
     if usecols is not None:
         usecols_dtype = lib.infer_dtype(usecols)
-        if usecols_dtype not in ('integer', 'string'):
-            raise ValueError(("The elements of 'usecols' "
-                              "must either be all strings "
-                              "or all integers"))
+
+        if usecols_dtype == 'mixed':
```

Comment:
Do you mean the  `usecols_dtype not in ('integer', 'string', 'unicode'):`?  I  took @sinhrks advice and by checking if the type is mixed and ensuring they are all string_type, it allows usecols of mixed encoding strings, and `test_usecols_with_mixed_encoding_strings` passes.

Or if you mean just have the `if not all(map(lambda x: isinstance(x, string_types), usecols))` line, then just having that would fail with mixed integer and strings, as those are `mixed-integer` dtype rather than just `mixed`.